src/coreclr/vm/ceemain.cpp

```diff
@@ -1329,12 +1329,6 @@ void STDMETHODCALLTYPE EEShutDownHelper(BOOL fIsDllUnloading)
             {
                 g_fEEShutDown |= ShutDown_Phase2;
 
-                // Shutdown finalizer before we suspend all background threads. Otherwise we
-                // never get to finalize anything.
-
-                // No longer process exceptions
-                g_fNoExceptions = true;
-
                 // <TODO>@TODO: This does things which shouldn't occur in part 2.  Namely,
                 // calling managed dll main callbacks (AppDomain::SignalProcessDetach), and
                 // RemoveAppDomainFromIPC.
```

Comment:
I am looking at what we do below and there seems to be more instances of questionable cleanup:
- CLRRemoveVectoredHandlers
- StubManager::TerminateStubManagers - it should just flush the debug-only log, but it should not be destroying the Crsts
- Interpreter::Terminate
-

Reply:
The Disassembler::StaticClose(); frees the disassembler shared library, it also sounds like something we should not do.

Reply:
StubManager::TerminateStubManagers - it actually doesn't do anything interesting. The whole log is just a SString instance that is never accessed in any way - there is a `StubManager::DbgGetLog` method, but nothing calls it.